### ISSUE TITLE
[Scala] taint analysis bug

### ISSUE THREAD

**Issue by pavkir** [semgrep/semgrep#4995]
**Describe the bug**
Taint analysis matches only first case #2

**To Reproduce**
rule: 
```yaml
rules:
  - id: test
    mode: taint
    pattern-sinks:
      - patterns:
          - pattern: new Hello(...)
    pattern-sources:
      - patterns:
          - pattern: $PARAM
          - pattern-inside: |
              def $FUNC(..., $PARAM: String, ...) = ...
    message: test
    languages: [scala]
    severity: WARNING
```
Code example:

```scala
class A() {
    // case1
    def func1(user: String) = {
        val test = new Hello(s "name - $user")

    }
    // case2 
    def func2(user: String) = {
        val test = new Hello(s "name - $user")
        random_call()
    }
}
```

**Expected behavior**
Both cases matched

**Screenshots**

**What is the priority of the bug to you?**
- [x] P1: important to fix or quite annoying

**Environment**
local semgrep 0.87.0

**Use case**
What will fixing this bug enable for you?

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-1139/[scala]-taint-analysis-bug.

**Comment by kopecs** [comment]
Fixed in #5658
